core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -478,6 +481,26 @@ public Result visit(Project e) {
     return builder.result();
   }
 
+  /**
+   * Whether the relNode is a join and whether its inputs have duplicate field names.
+   * For example, EMP JOIN DEPT, both of which have columns named DEPTNO
+   */
+  private boolean isJoinDuplicateName(RelNode relNode) {
+    if (relNode instanceof Join || relNode instanceof MultiJoin) {
+      List<RelNode> inputs = relNode.getInputs();
+      RelNode firstRel = inputs.get(0);
+      List<String> firstRelFiledNames = Lists.newArrayList(firstRel.getRowType().getFieldNames());
+      for (int i = 1; i < inputs.size(); i++) {
+        RelNode input = inputs.get(i);
```

Comment:
logic doesn't seem right if there are more than two inputs, and inputs 1 and 2 (0-based) have duplicate fields.

maybe write the field names into a list, and use `Util.isDistinct`

Reply:
What if the target SQL dialect is case-insensitive? For such a dialect "select t0.x, t1.X from t0, t1" would be a problem.

Maybe add a `boolean caseSensitive` parameter to `Util.isDistinct`, default true, and if it's false create a set with `new TreeSet<String>(String.CASE_INSENSITIVE_ORDER)`

Reply:
If it is a join with `USING` or `NATURAL JOIN` then the duplicate columns will have been filtered out. It's worth adding a test for `select * from emp join dept using (deptno)` and `select * from emp natural join dept` and make sure that `deptno` only occurs once in the output.